packages/backend/src/apps/formatter/common/transform-spec.ts

```diff
@@ -0,0 +1,32 @@
+import type {
+  IField,
+  IFieldDropdownOption,
+  IGlobalVariable,
+  IRawAction,
+} from '@plumber/types'
+
+type TransformFunction = (
+  $: IGlobalVariable,
+  valueToTransform: string,
+) => Awaited<ReturnType<IRawAction['run']>> | ReturnType<IRawAction['run']>
```

Comment:
i think i got a question if there is a need for `ReturnType<IRawAction['run']>` because does `Awaited` cover both?

Reply:
tbh i think I will remove `Awaited` because I tend to do `return await transformData(...)`